intake/source/base.py

```diff
@@ -53,7 +52,133 @@ class NoEntry(AttributeError):
     pass
 
 
-class DataSource(DictSerialiseMixin):
+class CacheMixin:
+    _cache = None
+
+    @property
+    def cache_dirs(self):
+        return [c._cache_dir for c in self.cache]
+
+    def set_cache_dir(self, cache_dir):
+        for c in self.cache:
+            c._cache_dir = make_path_posix(cache_dir)
+
+    @property
+    def cache(self):
+        if self._cache is None:
+            self._cache = make_caches(self.name, self.metadata.get('cache'),
+                                      catdir=self.metadata.get('catalog_dir',
+                                                             None),
+                                      storage_options=self.storage_options)
+        return self._cache
+
+    @cache.setter
+    def cache(self, csh):
+        self._cache = csh
+
+    def _get_cache(self, urlpath):
+        if len(self.cache) == 0:
+            return [urlpath]
+        return [c.load(urlpath) for c in self.cache]
+
+
+class GraphicsMixin:
```

Comment:
Can we call this `HoloviewsMixin`?

Reply:
Certainly! Or maybe HvPlot, to give the hint that this is plotting/gui-related?